oslo_vmware/rw_handles.py

```diff
@@ -310,30 +310,44 @@ def __init__(self, session, lease, url, file_handle,
 
     def _log_progress(self, progress):
         """Log data transfer progress."""
-        if (progress == 100 or (progress - self._last_logged_progress >=
-                                MIN_PROGRESS_DIFF_TO_LOG)):
-            LOG.debug("Data transfer progress is %d%%.", progress)
-            self._last_logged_progress = progress
+        if 'bytes_read' in progress:
+            metric = progress["bytes_read"]
+        else:
+            metric = progress["bytes_written"]
+
+        progress["metric"] = metric
+        time_diff = time.time() - self._last_progress_update
```

Comment:
You'd need something like we have in `_check_progress_stalled()`, because `self._last_progress_update` is 0 on first call, which would make your time difference multiple years ;)

Reply:
_last_progress_update is not initialized to time.time()